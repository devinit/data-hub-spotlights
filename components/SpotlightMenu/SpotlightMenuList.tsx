import React, { Children, cloneElement, isValidElement, FunctionComponent } from 'react';

interface SpotlightMenuListProps {
  active?: boolean;
  classNames?: string;
  onViewClick?: (event: React.MouseEvent<HTMLAnchorElement, MouseEvent>, title?: string) => void;
}

const SpotlightMenuList: FunctionComponent<SpotlightMenuListProps> = ({
  active,
  children,
  classNames,
  onViewClick
}) => {
  return (
    <ul className={classNames}>
      {Children.map(children, child => isValidElement(child) && cloneElement(child, { onView: onViewClick }))}
      <style jsx>{`
        display: ${active ? 'block !important' : 'none'};
        .countries-menu-list__content {
          max-height: 40vh;
        }
      `}</style>
    </ul>
  );
};

export { SpotlightMenuList };
