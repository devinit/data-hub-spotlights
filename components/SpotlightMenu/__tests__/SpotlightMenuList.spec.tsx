/**
 * @jest-environment jsdom
 */
import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import { SpotlightMenuList } from '../SpotlightMenuList';

describe('SpotlightMenuList', () => {
  test('renders correctly', () => {
    const renderer = TestRenderer.create(
      <SpotlightMenuList classNames="countries-menu-list__content" active={true} />
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('It is not rendered when active prop is false', () => {
    const renderer = TestRenderer.create(
      <SpotlightMenuList active={false} classNames="countries-menu-list__content" />
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });

  test('renders children correctly', () => {
    const renderer = TestRenderer.create(
      <SpotlightMenuList classNames="countries-menu-list__content" active={true}>
        <div>Testing</div>
      </SpotlightMenuList>
    ).toJSON();

    expect(renderer).toMatchSnapshot();
  });
});
