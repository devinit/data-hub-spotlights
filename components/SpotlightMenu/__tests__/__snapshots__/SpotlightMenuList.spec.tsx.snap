// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpotlightMenuList It is not rendered when active prop is false 1`] = `
<ul
  className="countries-menu-list__content"
>
  <style
    jsx={true}
  >
    
        display: none;
        text-transform: capitalize;
      
  </style>
</ul>
`;

exports[`SpotlightMenuList renders children correctly 1`] = `
<ul
  className="countries-menu-list__content"
>
  <div>
    Testing
  </div>
  <style
    jsx={true}
  >
    
        display: block !important;
        text-transform: capitalize;
      
  </style>
</ul>
`;

exports[`SpotlightMenuList renders correctly 1`] = `
<ul
  className="countries-menu-list__content"
>
  <style
    jsx={true}
  >
    
        display: block !important;
        text-transform: capitalize;
      
  </style>
</ul>
`;
